### ISSUE TITLE
Synthèse et occurrence d'absence

### ISSUE THREAD

**Issue by camillemonchicourt** [PnX-SI/GeoNature#521]
Certains modules peuvent produire des occurrences d'absence. Les modules de suivi notamment, mais aussi Occtax pour ceux qui souhaitent afficher le champs "Statut de l'observation" (voir https://github.com/PnX-SI/GeoNature/issues/188).

Ces occurrences d'observation peuvent être reportées dans la Synthèse. Celle-ci étant basée sur le standard Occurrence de taxon, elles contiennent aussi la nomenclature "Statut de l'observation" (``ref_nomenclatures.bib_nomenclatures_types = 18`` / https://github.com/PnX-SI/Nomenclature-api-module/blob/master/data/data_nomenclatures.sql#L31) avec les 3 valeurs du SINP correspondantes (https://github.com/PnX-SI/Nomenclature-api-module/blob/master/data/data_nomenclatures.sql#L211-L213) définies ici : http://standards-sinp.mnhn.fr/nomenclature/18-statut-dobservation-statutobservation-16-08-2016/

Pour leur répercussion au niveau de la Synthèse, différentes approches sont possibles : 
- Faire en sorte que le trigger alimentant la Synthèse depuis le module ne renvoie pas les absences dans la Synthèse. C'est déjà le cas dans Occtax de manière générique, comme toutes les autres nomenclatures.
- Intégrer les données d'absence dans la Synthèse depuis le module, en renseignant bien la nomenclature "Statut d'observation"

Etant donné que le SINP prévoit les occurrences d'absence (malgré leur faible valeur sorties de leur protocole), c'est plutôt la deuxième solution qui serait à retenir.
Actuellement la valeur de cette nomenclature est affichée dans la Synthèse, comme toutes les autres nomenclatures au niveau de la fiche Détail d'une occurrence. 
Et elle est filtrable de manière générique comme toutes les autres nomenclatures. 
Il n'y a pas de traitement spécifique de cette nomenclature. 

Côté GeoNature-atlas, il ne semble pas opportun d'intégrer les données d'absence donc il suffira de compléter la vue ``atlas.vm_observations`` en ajoutant dans son ``where`` que les observations où "Statut d'observation = Présent" (voir https://github.com/PnEcrins/GeoNature-atlas/issues/165)

**Comment by gildeluermoz** [comment]
Faire des prospections concernant un taxon, ce n'est pas du contact occasionnel (occtax).
Ok, il n'y a pas d'outil pour saisir ce genre de données pour le moment, mais ce n'est pas une bonne raison pour tout mélanger. La valeur d'absence est toute relative et totalement dépendante du protocole. Qui peut affirmer à 100% qu'un taxon est absent ?
Mais comme le dit Camille, chacun est libre de le faire et d'en assumer toutes les répercussions dans les interfaces qui doivent cependant rester cohérentes.

Par contre, ok pour la réflexion à conduire sur le comportement de la synthèse vis à vis de ces données d'absence qui peuvent arriver d'un module produisant ce genre de données de manière protocolée (où la valeur d'absence est bien définie par le protocole).

Voici une proposition de traitement en interface : 
* ces données ne devraient pas s'afficher avec les données de présence (en tout cas pas par défaut)
* exclure de l'affichage par défaut les données d'absence
    * l'utilisateurs coche un bouton 'afficher les données d'absence'
    * les données d'absence s'affichent soit seules, soit au milieu des autres mais avec une identification dédiée

Si ces données sont dans la synthèse, il faut étendre la réflexion sur le module export et le module validation...

**Comment by camillemonchicourt** [comment]
La première question est de savoir si on intègre les données d'absence dans la Synthèse ou si on ne les laisse que dans leurs modules, seul endroit où elles ont vraiment du sens, car contextualisés dans leur protocole. 

Ne pas les mettre dans la Synthèse permet d'éviter des erreurs d'interprétation et de confusion dans la Synthèse où l'utilisateur ne s'attendra pas à ce qu'une ligne corresponde à une occurrence qu'on n'a pas vu.

Mais ne pas les mettre dans la Synthèse les exclut des exports et diffusion vers les partenaires et le SINP (qui les prévoit dans le standard Occurrences de taxon) qui se baseront majoritairement sur la Synthèse.

La solution intermédiaire est de les intégrer mais avec un traitement spécifique pour ne pas les afficher par défaut, les afficher d'une autre couleur ou autre. 

A voir si c'est utile au regard de leur pertinence dans la Synthèse.

**Comment by pierrelapenu** [comment]
Effectivement, personne ne peut affirmer à 100% qu'un taxon est absent. On sait même que certains taxons présents sont très discrets à certaines périodes de l'année ou certains moments de la journée. En revanche, je maintiens qu'il peut être intéressant dans certaines circonstances de noter que lors d'un passage à un endroit précis, on a pas vu un taxon donné (parce qu'on sait qu'il peut y être présent, qu'il semble que le milieu soit favorable, qu'on souhaite justement mieux connaitre les périodes de présence, ...). Afin de "ne pas tout mélanger", effectivement, il faut être prudents dans la gestion de ces données. Les solutions proposées me paraissent très adaptée si ce n'est pas trop difficile techniquement : Une case à cocher pour choisir explicitement si on affiche ou on exporte ces données, une représentation différente si possible et pas de représentation dans l'atlas ou, me semble-t-il, ce n'est pas vraiment le lieu et ce serait trop lourd et source de confusion.

**Comment by camillemonchicourt** [comment]
Oui

**Comment by RomainBaghi** [comment]
Bonjour, 

Je vous rejoins sur cette discussion, effectivement il est très intéressant de pouvoir noter qu'une prospection pour une espèce n'a rien donnée (c'est comme cela que je vois la donnée d'absence).
En testant la saisie d'une donnée de ce type, on m'a fait remonté qu'il était impossible de saisir un dénombrement à 0. Par défaut le dénombrement est à 1. Je pense qu'il faut mettre le dénombrement à 0 automatiquement quand c'est une donnée de ce type.
merci

**Comment by camillemonchicourt** [comment]
Autoriser 0 comme valeur du dénombrement semble OK.
Mais mettre 0 automatiquement quand on indique qu'il s'agit d'une Absence est moins souhaitable car cela intègre de la spécificité dans le module.

**Comment by RomainBaghi** [comment]
Ok, je comprends bien.
J'ai juste peur que les observateurs qui renseignent le statut d'obs. à "Non Observé", n'aillent pas plus loin dans le déroulé du formulaire et donc valide l'obs sans aller voir ce qui a été noté par défaut pour la partie dénombrement. Mais je peux gérer ça en base sur mon instance gn.

**Comment by camillemonchicourt** [comment]
Je comprends. 
@TheoLechemia vient de commité la possibilité de mettre "0" comme dénombrement : https://github.com/PnX-SI/GeoNature/commit/eb43138b50fbcf04456ef92187d67f3cc6215ca3

**Comment by camillemonchicourt** [comment]
Dans la version 2.12.0 il est possible de renseigner de nouveaux paramètres du module Synthèse permettant de définir des filtres activés par défaut et leur valeur associée : https://github.com/PnX-SI/GeoNature/pull/2262/files#diff-5b0da6f626536c3f89d4ccd93ac5892ac399fb7a9ab1944fb801efa53696cd58

Ces données d'absence sont bien intégrées et présentes dans le module Synthèse, mais donc filtrables par défaut.

Actuellement, elles ne sont pas spécifiquement distinguées dans la liste des observations, ni dans les fiches de chaque observation.

**Comment by DonovanMaillard** [comment]
Rien de plus de planifié sur ce sujet, déjà largement pris en charge aujourd'hui avec la saisie possible dans occtax, l'envoi en synthèse, le requêtage et l'export possible avec la nomenclature dédiée et depuis 2.12 des filtres par défaut pour les exclure des résultats sauf demande contraire de l'utilisateur.

**Comment by camillemonchicourt** [comment]
Et en complement, un développement est en cours et devrait intégrer une prochaine version, permettant de distinguer les observations dans le modules Synthèse, sur la base de critères des nomenclatures, qui peut justement être celle identifiant Présence/Absence : https://github.com/PnX-SI/GeoNature/issues/3258
